examples/qaoa/parse_raw_sat.py

```diff
@@ -0,0 +1,215 @@
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A module to parse maxSAT problems from files."""
+
+import argparse
+from collections import namedtuple
+import os
+import re
+
+import numpy as np
+
+from qubo import QuboProblem
+
+CONSTRAINT_RE = re.compile(r'R[0-9]+\:(-?(?:X[0-9]+[+-])*X[0-9]+)>=(-?[0-9]+)')
+VARIABLE_RE = re.compile(r'[+-]?X[0-9]+')
+
+
+_Sat = namedtuple('_Sat', ['clauses', 'num_vars', 'source_filename'])
+
+
+class Sat(_Sat):
+    """A named tuple that represents a max-SAT problem.
+
+    Attributes:
+        clauses: list of Clauses
+        num_vars: amount of variables
+        source_filame: the path to the source file (Daimler's format)
+    """
+
+    def _check_clause(self, clause, solution):
+        for el in clause:
+            s = solution[abs(el) - 1]
+            if el > 0 and s == 1:
+                return True
+            elif el < 0 and s == 0:
```

Comment:
ack

Reply:
I actually don't think the `return (el > 0 and s ==1) or (el < 0 and s == 0)` would work here, since we are iterating and returning True if we've found smth during the iteration.

Reply:
That's exactly what the combined return would do... It will return true if either condition is true which is the same as the logic you have already.

Reply:
If it's complicated enough for a discussion, it should have a unit test for this function.

Reply:
no it's not
and there is a test for this function